src/main/webapp/app/exercises/shared/exercise-hint/shared/solution-entry.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, OnInit, ViewChild, Input, Output, EventEmitter } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Observable } from 'rxjs';
+
+import { AceEditorComponent } from 'app/shared/markdown-editor/ace-editor/ace-editor.component';
+import { faTimes } from '@fortawesome/free-solid-svg-icons';
+import { ProgrammingExerciseSolutionEntry } from 'app/entities/hestia/programming-exercise-solution-entry.model';
+
+@Component({
+    selector: 'jhi-solution-entry',
+    templateUrl: './solution-entry.component.html',
+})
+export class SolutionEntryComponent implements OnInit {
+    @ViewChild('editor', { static: true })
+    editor: AceEditorComponent;
+
+    @Input()
+    solutionEntry: ProgrammingExerciseSolutionEntry;
+    @Input()
+    enableEditing: boolean;
+    @Input()
+    dialogError: Observable<string>;
+
+    @Output()
+    onRemoveEntry: EventEmitter<any> = new EventEmitter();
+
+    faTimes = faTimes;
+
+    constructor(protected route: ActivatedRoute) {}
+
+    ngOnInit() {
+        this.setupEditor();
+    }
+
+    emitRemovalEvent() {
+        this.onRemoveEntry.emit();
+    }
+
+    private setupEditor() {
+        const line = this.solutionEntry.line;
+
+        this.editor.setTheme('dreamweaver');
```

Comment:
Please subscribe to the ``ThemeService`` to set this theme dynamically. Please have a look at other examples where the code editor is used to learn how.

Reply:
Implemented in 194dc2d.